kernel/common/file.rb

```diff
@@ -1195,6 +1195,21 @@ def self.sticky?(file_name)
     return false
   end
 
+  def self.mkfifo(file_name, mode = 0666)
+    raise NotImplementedError, "mkfifo is not implemented on this platform" unless Rubinius::HAVE_MKFIFO
```

Comment:
I think it's better to only define this method when `mkfifo` is actually present, opposed to checking for it on every call.